File: app/src/main/java/com/youtube/sorcjc/redemnorte/io/MyApiService.java
package com.youtube.sorcjc.redemnorte.io;

import com.youtube.sorcjc.redemnorte.io.response.BienResponse;
import com.youtube.sorcjc.redemnorte.io.response.BienesResponse;
import com.youtube.sorcjc.redemnorte.io.response.ByOldCodeResponse;
import com.youtube.sorcjc.redemnorte.io.response.ByPatrimonialResponse;
import com.youtube.sorcjc.redemnorte.io.response.HojaResponse;
import com.youtube.sorcjc.redemnorte.io.response.ResponsableResponse;
import com.youtube.sorcjc.redemnorte.io.response.SimpleResponse;
import com.youtube.sorcjc.redemnorte.model.Sheet;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface MyApiService {

    @GET("responsables")
    Call<ResponsableResponse> getResponsables();

    @FormUrlEncoded
    @POST("registrar-foto")
    Call<SimpleResponse> postPhoto(
            @Field("image") String base64, @Field("extension") String extension,
            @Field("hoja_id") String hoja_id, @Field("QR_code") String QR_code);

    @GET("sheets")
    Call<ArrayList<Sheet>> getSheets(@Query("dni") String dni);

    @GET("/sheets/1")
    Call<HojaResponse> getSheet(@Query("sheet_id") String sheet_id);

    @GET("items")
    Call<BienesResponse> getItems(@Query("sheet_id") String sheet_id);

    @GET("bien")
    Call<BienResponse> getItem(@Query("hoja_id") String hoja_id, @Query("QR_code") String QR_code);

    @GET("login")
    Call<SimpleResponse> getLogin(@Query("username") String username, @Query("password") String password);

    @FormUrlEncoded
    @POST("sheets")
    Call<SimpleResponse> storeSheet(
            @Field("id") String id,
            @Field("local") String local,
            @Field("ubicacion") String ubicacion,
            @Field("responsable") String responsable,
            @Field("cargo") String cargo,
            @Field("oficina") String oficina,
            @Field("ambiente") String ambiente,
            @Field("area") String area,
            @Field("activo") String activo,
            @Field("observacion") String observacion,
            @Field("inventariador") String inventariador
    );

    @FormUrlEncoded
    @POST("editar-hoja")
    Call<SimpleResponse> updateSheet(
            @Field("id") String id,
            @Field("local") String local,
            @Field("ubicacion") String ubicacion,
            @Field("responsable") String responsable,
            @Field("cargo") String cargo,
            @Field("oficina") String oficina,
            @Field("ambiente") String ambiente,
            @Field("area") String area,
            @Field("activo") String activo,
            @Field("observacion") String observacion
    );

    @GET("check-qr")
    Call<SimpleResponse> getCheckQr(@Query("qr_code") String QR_code);

    @GET("take-by-patrimonial")
    Call<ByPatrimonialResponse> getByPatrimonial(@Query("patrimonial") String patrimonial);

    @GET("take-by-old-code")
    Call<ByOldCodeResponse> getByOldCode(@Query("year") String year, @Query("code") String code);

    @FormUrlEncoded
    @POST("items")
    Call<SimpleResponse> storeItem(
            @Field("hoja_id") String hoja_id,
            @Field("QR_code") String QR_code,
            @Field("patrimonial_code") String patrimonial_code,
            @Field("old_code") String old_code,
            @Field("old_year") String old_year,

            @Field("denominacion") String denominacion,
            @Field("marca") String marca,
            @Field("modelo") String modelo,
            @Field("serie") String serie,
            @Field("color") String color,

            @Field("largo") String largo,
            @Field("ancho") String ancho,
            @Field("alto") String alto,

            @Field("condicion") String condicion,
            @Field("etiquetado") String etiquetado,
            @Field("operativo") String operativo,
            @Field("observacion") String observacion
    );

    @FormUrlEncoded
    @POST("items/update")
    Call<SimpleResponse> updateItem(
            @Field("hoja_id") String hoja_id,
            @Field("QR_code") String QR_code,
            @Field("patrimonial_code") String patrimonial_code,
            @Field("old_code") String old_code,
            @Field("old_year") String old_year,

            @Field("denominacion") String denominacion,
            @Field("marca") String marca,
            @Field("modelo") String modelo,
            @Field("serie") String serie,
            @Field("color") String color,

            @Field("largo") String largo,
            @Field("ancho") String ancho,
            @Field("alto") String alto,

            @Field("condicion") String condicion,
            @Field("etiquetado") String etiquetado,
            @Field("operativo") String operativo,
            @Field("observacion") String observacion
    );
}

File: app/src/main/java/com/youtube/sorcjc/redemnorte/io/MyApiService.kt
package com.youtube.sorcjc.redemnorte.io

import com.youtube.sorcjc.redemnorte.io.response.*
import com.youtube.sorcjc.redemnorte.model.Item
import com.youtube.sorcjc.redemnorte.model.ResponsibleUser
import com.youtube.sorcjc.redemnorte.model.Sheet
import com.youtube.sorcjc.redemnorte.model.User
import retrofit2.Call
import retrofit2.http.*
import java.util.*

interface MyApiService {
    @POST("login")
    @Headers("Accept: application/json")
    fun postLogin(
            @Query("username") username: String?,
            @Query("password") password: String?
    ): Call<User>

    @GET("public-data")
    @Headers("Accept: application/json")
    fun getPublicData(): Call<PublicDataResponse>

    @GET("sheets")
    @Headers("Accept: application/json")
    fun getSheets(@Query("user_id") userId: Int): Call<ArrayList<Sheet>>

    @GET("sheets/{sheetId}")
    @Headers("Accept: application/json")
    fun getSheet(@Path("sheetId") sheetId: Int): Call<Sheet>

    @GET("sheets/{sheetId}/items")
    @Headers("Accept: application/json")
    fun getItems(@Path("sheetId") sheetId: Int): Call<ArrayList<Item>>

    @GET("items/{itemId}")
    @Headers("Accept: application/json")
    fun getItem(@Path("itemId") itemId: Int): Call<Item>

    @FormUrlEncoded
    @POST("items/{itemId}/photos")
    @Headers("Accept: application/json")
    fun postPhoto(
            @Path("itemId") itemId: Int,
            @Field("image") base64: String?,
            @Field("extension") extension: String?
    ): Call<Item>

    @FormUrlEncoded
    @POST("sheets")
    @Headers("Accept: application/json")
    fun storeSheet(
            @Field("place") place: String?,
            @Field("location") location: String?,
            @Field("responsible_user") responsibleName: String?,
            @Field("position") position: String?,
            @Field("office") office: String?,
            @Field("ambient") ambient: String?,
            @Field("area") area: String?,
            @Field("pending") pending: Boolean,
            @Field("observation") observation: String?,
            @Field("author") userId: Int
    ): Call<Sheet>

    @FormUrlEncoded
    @PUT("sheets/{sheetId}")
    @Headers("Accept: application/json")
    fun updateSheet(
            @Path("sheetId") sheetId: Int,
            @Field("place") place: String?,
            @Field("location") location: String?,
            @Field("responsible_user") responsibleName: String?,
            @Field("position") position: String?,
            @Field("office") office: String?,
            @Field("ambient") ambient: String?,
            @Field("area") area: String?,
            @Field("pending") pending: Boolean,
            @Field("observation") observation: String?
    ): Call<Sheet>

    @GET("check-qr")
    @Headers("Accept: application/json")
    fun getCheckQr(@Query("qr_code") qrCode: String?): Call<SimpleResponse>

    @GET("expected-data/search")
    @Headers("Accept: application/json")
    fun getByPatrimonial(@Query("code") code: String): Call<Item>

    @GET("old-data/search")
    @Headers("Accept: application/json")
    fun getByOldCode(@Query("year") year: String, @Query("code") code: String): Call<Item>

    @FormUrlEncoded
    @POST("items")
    @Headers("Accept: application/json")
    fun storeItem(
            @Field("sheet_id") sheetId: Int,
            @Field("inventory_code") qrCode: String?,
            @Field("patrimonial_code") patrimonial_code: String?,
            @Field("old_code") old_code: String?,
            @Field("old_year") old_year: String?,
            @Field("denomination") denomination: String?,
            @Field("brand") brand: String?,
            @Field("model") model: String?,
            @Field("series") series: String?,
            @Field("color") color: String?,
            @Field("length") length: String?,
            @Field("width") width: String?,
            @Field("height") height: String?,
            @Field("status") status: Int,
            @Field("labeled") labeled: Boolean,
            @Field("operative") operative: Boolean,
            @Field("observation") observation: String?
    ): Call<Item>

    @FormUrlEncoded
    @PUT("items/{itemId}")
    @Headers("Accept: application/json")
    fun updateItem(
            @Field("itemId") itemId: Int,
            @Field("inventory_code") qrCode: String?,
            @Field("patrimonial_code") patrimonial_code: String?,
            @Field("old_code") old_code: String?,
            @Field("old_year") old_year: String?,
            @Field("denomination") denomination: String?,
            @Field("brand") brand: String?,
            @Field("model") model: String?,
            @Field("series") series: String?,
            @Field("color") color: String?,
            @Field("length") length: String?,
            @Field("width") width: String?,
            @Field("height") height: String?,
            @Field("status") status: Int,
            @Field("labeled") labeled: Boolean,
            @Field("operative") operative: Boolean,
            @Field("observation") observation: String?
    ): Call<Item>
}